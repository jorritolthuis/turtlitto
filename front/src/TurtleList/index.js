import React from "react";
import PropTypes from "prop-types";
import { Row, Col } from "react-flexbox-grid";

import Turtle from "../Turtle";

const TurtleList = props => {
  const { turtles } = props;
  return (
    <Row>
      {turtles.map((turtle, index) => (
        <Col xs={12} md={6} key={index}>
          <Turtle key={index} turtle={turtle} editable />
        </Col>
      ))}
    </Row>
  );
};

TurtleList.propTypes = {
  turtles: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      enabled: PropTypes.bool.isRequired,
      batteryvoltage: PropTypes.number.isRequired,
      homegoal: PropTypes.string.isRequired,
      role: PropTypes.string.isRequired,
      teamcolor: PropTypes.string.isRequired
    })
  ).isRequired
};

export default TurtleList;
