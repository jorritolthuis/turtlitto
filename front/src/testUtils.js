import React from "react";
import { ThemeProvider } from "styled-components";
import theme from "./theme";
import { shallow, mount } from "enzyme";

export const shallowWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  });
};
