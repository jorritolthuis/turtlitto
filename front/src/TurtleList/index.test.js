import React from "react";
import TurtleList from ".";
import { shallowWithTheme, mountWithTheme } from "../testUtils";

describe("TurtleList", () => {
  it("should match snapshot", () => {
    const wrapper = shallowWithTheme(<TurtleList turtles={{}} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render all turtles", () => {
    const wrapper = mountWithTheme(
      <TurtleList
        turtles={{
          1: {
            enabled: true,
            batteryvoltage: 66,
            homegoal: "yellow",
            role: "inactive",
            teamcolor: "magenta"
          },
          2: {
            enabled: false,
            batteryvoltage: 42,
            homegoal: "yellow",
            role: "inactive",
            teamcolor: "magenta"
          }
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
