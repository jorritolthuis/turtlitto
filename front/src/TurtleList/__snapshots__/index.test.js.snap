// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurtleList should match snapshot 1`] = `
<Row>
  <Col
    key="0"
    md={6}
    xs={12}
  >
    <Turtle
      editable={true}
      key="1"
      turtle={
        Object {
          "batteryvoltage": 66,
          "enabled": true,
          "homegoal": "Yellow home",
          "id": 1,
          "role": "INACTIVE",
          "teamcolor": "Magenta",
        }
      }
    />
  </Col>
  <Col
    key="1"
    md={6}
    xs={12}
  >
    <Turtle
      editable={true}
      key="2"
      turtle={
        Object {
          "batteryvoltage": 42,
          "enabled": false,
          "homegoal": "Yellow home",
          "id": 2,
          "role": "INACTIVE",
          "teamcolor": "Magenta",
        }
      }
    />
  </Col>
</Row>
`;
